File: src/code/js/Q122.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    var max_profit = 0;
    for (var i = 1; i < prices.length; i++) {
        if (prices[i] > prices[i - 1]) {
            max_profit += (prices[i] - prices[i - 1]);
        }
    }
    return max_profit;
};

File: src/code/java/Q122.java
package code.java;

public class Q122 {
    public int maxProfit(int[] prices) {
        int max_profit = 0;
        for (int i = 1; i < prices.length; i++) {
            max_profit += Math.max(prices[i] - prices[i - 1], 0);
        }
        return max_profit;
    }
}
